pkg/project/manager.go

```diff
@@ -139,23 +140,23 @@ func (pm ManagerImpl) CreateProject(ctx context.Context, request CreateProjectRe
 
 		project := model.Project{Id: projectId, Path: projectPath, Config: model.Config{DevContainerConfig: devContainerConfig}, ContainerId: containerId}
 
-		opts := []files.ListFileOption{
-			files.ListFilesWithContent(),
-		}
-
-		files, err := pm.fileManager.ListFiles(model.NewContextWithProject(context.Background(), &project), afero.NewBasePathFs(afero.NewOsFs(), projectPath), opts...)
-		if err != nil {
-			log.Error().Err(err).Msg("Failed to list files")
-			removeProjectDir(projectPath)
-			c <- result.Failure[model.Project](fmt.Errorf("Failed to list files: %w", err))
-			return
+		var languages []lsp.LanguageId
+		if request.Languages != nil {
+			languages = request.Languages
+		} else {
+			languages, err = pm.detectLanguages(project)
+			if err != nil {
+				log.Error().Err(err).Msg("Failed to detect project languages")
+				removeProjectDir(projectPath)
```

Comment:
Also do we really need to write to a channel?

Reply:
We run project creation in a goruotine because it takes a while. That's why we used channels. But I'm open to any suggestions here. This one was my first goroutine, so it's far from ideal :)

Reply:
`result := <-h.Manager.CreateProject(r.Context(), request)` in handler is equivalent to method invocation as far as I can tell. But let's take it outside this PR.